cylc/flow/install.py

```diff
@@ -213,19 +213,12 @@ def reinstall_workflow(
         dry_run=dry_run,
     )
 
-    # Add '+++' to -out-format to mark lines passed through formatter.
-    rsync_cmd.append('--out-format=+++%o %n%L+++')
-
     # Run rsync command:
     reinstall_log.info(cli_format(rsync_cmd))
     LOG.debug(cli_format(rsync_cmd))
     proc = Popen(rsync_cmd, stdout=PIPE, stderr=PIPE, text=True)  # nosec
     # * command is constructed via internal interface
-    stdout, stderr = proc.communicate()
-
-    # Strip unwanted output.
-    stdout = ('\n'.join(re.findall(r'\+\+\+(.*)\+\+\+', stdout))).strip()
```

Comment:
This seemed to be duplicating the work of `format_rsync_output()`, but removing all `cannot delete non-empty directory:` lines, not just the `opt` dir

Reply:
The point of the `+++` prefix/suffix will have been to separate intended `rsync` output from any unexpected "junk" that might be output by the command / shell, ensuring that we only display what we need, even if the output differs from expectation in other ways.

Reply:
From the context I could gather in https://github.com/cylc/cylc-flow/pull/5125, the only unwanted output that was being targeted was `cannot delete non-empty directory:`.

Reply:
This `rsync` command doesn't just apply to the dry-run check (which is passed through`format_rsync_output`). And this prefix stripped out other potential output.

Reply:
https://github.com/cylc/cylc-flow/blob/48d3d3898fd06c0f25c3214b220389a336c01fc3/cylc/flow/install.py#L226-L227
is at odds with
https://github.com/cylc/cylc-flow/blob/48d3d3898fd06c0f25c3214b220389a336c01fc3/cylc/flow/scripts/reinstall.py#L346-L348

The former seems to have been implemented to cover `cannot delete non-empty directory:`, there was no mention of any other potential rsync output in #5125. And the latter seems to have been implemented to deliberately include other potential rsync output in case the rsync implementation results in non-standard output format.

Hence I am not sure we should be stripping other potential rsync output.

Reply:
It's not at odds.

* The `+++` prefix ensures that only the intended output lines get through.
* The `else` handles output lines that don't match `send` or `del.`.